### PR TITLE
Sig Instrumentation Spotlight

### PR DESCRIPTION
**Pull request by imnmo**
HackMd: https://hackmd.io/@TyxQSpShS_mQBOxa_4vq5A/BJWfsol_o/edit

### CODE REVIEW

**Comment by imnmo** [comment]
@kaslin @fsmunoz PTAL

**Comment by jberkus** [comment]
Important question: is this going only on the Contributor blog, or should it go on the main Kubernetes blog?  It's looking like something for the main blog.

**Comment by sftim** [comment]
We usually mirror SIG spotlight articles onto the main blog (which means: once this is edited enough to be reviewable, /hold _this_ PR and open a similar PR against k/website, so we can get them in sync to publish on the same day).

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+layout: blog
+title: "Spotlight on SIG Instrumentation"
+slug: sig-instrumentation-spotlight
+date: 2023-01-31
+canonicalUrl: https://www.kubernetes.dev/blog/2023/01/31/sig-instrumentation-spotlight-2023/
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:6]: Don't set a `canonicalUrl` for **this** PR. However, the article on https://blog.k8s.io/ does need that front matter variable set.

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+**Author**: Imran Noor Mohamed(Delivery Hero)
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:9]: For this site, AIUI we prefer to use the front matter to set an author. See other articles such as https://github.com/kubernetes/contributor-site/blame/master/content/en/blog/2022/sig-storage-spotlight.md#L6 for an example.

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Observability requires the right data at the right time for the right consumer (human or piece of software) to make the right decision. In the context of Kubernetes, having best practices for cluster observability across all kubernetes components is crucial. 
+
+SIG Instrumentation helps to address this issue by providing best practices and tools that all other SIG uses to instrument kubernetes components like the *Kube API server*, *scheduler*, and *controller manager*.
+
+In this SIG Instrumentation spotlight, [Imran Noor Mohamed](https://www.linkedin.com/in/imrannoormohamed/), SIG Contribex-Comms tech lead talked with [Elana Hashman](https://twitter.com/ehashdn), and [Han Kang](https://www.linkedin.com/in/hankang), Chairs of SIG instrumentation, on how the SIG is organized, what are the current challenges and how anyone can get involved and contribute.
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:15]:
```suggestion
In this SIG Instrumentation spotlight, [Imran Noor Mohamed](https://www.linkedin.com/in/imrannoormohamed/), SIG ContribEx-Comms tech lead talked with [Elana Hashman](https://twitter.com/ehashdn), and [Han Kang](https://www.linkedin.com/in/hankang), chairs of SIG Instrumentation, on how the SIG is organized, what are the current challenges and how anyone can get involved and contribute. 
``

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+**Imran (INM)**: Hello, thank you for the opportunity of learning more about SIG Instrumentation. Could you tell us a bit about yourself, your role, and how you got involved in SIG Instrumentation?
+
+**Han (HK)**: I started in SIG Instrumentation in 2018, and became a chair in 2020. I primarily got involved with SIG instrumentation due to a number of upstream issues with metrics which ended up affecting GKE in bad ways. As a result, we ended up launching an initiative to stabilize our metrics and make metrics a proper API.
+
+**Elana (EH)**: I also joined SIG Instrumentation in 2018 and became a chair at the same time as Han. I was working as a site reliability engineer (SRE) on bare metal Kubernetes clusters and was working to build out our observability stack. I encountered some issues with label joins where Kubernetes metrics didn’t match Kube-state-metrics (KSM) and started participating in SIG meetings to improve things. I helped test performance improvements to Kube-state-metrics and ultimately coauthored a KEP for overhauling metrics in the 1.14 release to improve usability.
+
+**Imran (INM)**: Interesting! Does that mean SIG-Instrumentation involves a lot of plumbing?
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:25]:
```suggestion
**Imran (INM)**: Interesting! Does that mean SIG Instrumentation involves a lot of plumbing?
```

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+**Han (HK)**: I wouldn’t say it involves a ton of plumbing, though it does touch basically every code base. We have our own dedicated directories for our metrics, logs, and tracing frameworks which we tend to work out of primarily. We do have to interact with other SIGs in order to propagate our changes which makes us more of a horizontal SIG.
+
+**Imran (INM)**: Speaking about interaction and coordination with other SIG could you describe how the SIG is organized?
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:29]: Was this actually:
```suggestion
**Imran (INM)**: Speaking about interaction and coordination with other SIGs could you describe how the SIG is organized?
```

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+slug: sig-instrumentation-spotlight
+date: 2023-01-31
+canonicalUrl: https://www.kubernetes.dev/blog/2023/01/31/sig-instrumentation-spotlight-2023/
+---
+
+**Author**: Imran Noor Mohamed(Delivery Hero)
+
+Observability requires the right data at the right time for the right consumer (human or piece of software) to make the right decision. In the context of Kubernetes, having best practices for cluster observability across all kubernetes components is crucial. 
+
+SIG Instrumentation helps to address this issue by providing best practices and tools that all other SIG uses to instrument kubernetes components like the *Kube API server*, *scheduler*, and *controller manager*.
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:13]:
```suggestion
SIG Instrumentation helps to address this issue by providing best practices and tools that all other SIGs use to instrument Kubernetes components–like the *API server*, *scheduler*, *kubelet* and *controller manager*.
```
?

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+layout: blog
+title: "Spotlight on SIG Instrumentation"
+slug: sig-instrumentation-spotlight
+date: 2023-01-31
+canonicalUrl: https://www.kubernetes.dev/blog/2023/01/31/sig-instrumentation-spotlight-2023/
+---
+
+**Author**: Imran Noor Mohamed(Delivery Hero)
+
+Observability requires the right data at the right time for the right consumer (human or piece of software) to make the right decision. In the context of Kubernetes, having best practices for cluster observability across all kubernetes components is crucial.
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:11]:
```suggestion
Observability requires the right data at the right time for the right consumer (human or piece of software) to make the right decision. In the context of Kubernetes, having best practices for cluster observability across all components is crucial.
```

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
+---
+layout: blog
+title: "Spotlight on SIG Instrumentation"
+slug: sig-instrumentation-spotlight
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:4]:
```suggestion
slug: sig-instrumentation-spotlight-2023
```

**Review by sftim — COMMENTED** [review]
Thanks for the PR.

I can offer some feedback.

Overall:
- sentence case headings please
- `kube-state-metrics` not Kube State Metrics.
- SIG Instrumentation not SIG-Instrumentation
-

File: content/en/blog/2023/sig-instrumentation-spotlight.md (hunk 3 above)
Comment (imnmo) [reply content/en/blog/2023/sig-instrumentation-spotlight.md:15]: done

File: content/en/blog/2023/sig-instrumentation-spotlight.md (hunk 4 above)
Comment (imnmo) [reply content/en/blog/2023/sig-instrumentation-spotlight.md:25]: done

File: content/en/blog/2023/sig-instrumentation-spotlight.md (hunk 5 above)
Comment (imnmo) [reply content/en/blog/2023/sig-instrumentation-spotlight.md:29]: yes, that was the intention. I got confused between plural and singular forms.

File: content/en/blog/2023/sig-instrumentation-spotlight.md (hunk 6 above)
Comment (imnmo) [reply content/en/blog/2023/sig-instrumentation-spotlight.md:13]: done

**Comment by imnmo** [comment]
> Thanks for the PR.
> 
> I can offer some feedback.
> 
> Overall:
> 
> * sentence case headings please
> * `kube-state-metrics` not Kube State Metrics.
> * SIG Instrumentation not SIG-Instrumentation

Thanks a lot, @sftim I am done with fixing it. Will upload a new patch.

**Comment by imnmo** [comment]
> Important question: is this going only on the Contributor blog, or should it go on the main Kubernetes blog? It's looking like something for the main blog.

Quoting @sftim once this is ready to merge stage I will open a new PR within K8s.io to make sync and review easier

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+## Current status and ongoing challenges
+
+**Imran (INM)**: For release [1.26](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.26) we can see that there are a relevant number of metrics, logs, and tracing [KEPs](https://www.k8s.dev/resources/keps/) in the pipeline. Would you like to point out important things for last release (maybe alpha & stable milestone candidates?)
+
+**Han (HK)**: We can now generate documentation for every single metric in the main Kubernetes code base! We have a pretty fancy static analysis pipeline that enables this functionality. We’ve also added feature metrics so that you can look at your metrics to determine which features are enabled in your cluster at a given time. Lastly, we added a component-sli endpoint, which should make it easy for people to create availability SLOs for *control-plane* components.
```
Comment (fsmunoz) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:43]: Link to https://kubernetes.io/docs/reference/instrumentation/metrics/ ?

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
+---
+layout: blog
+title: "Spotlight on SIG Instrumentation"
+slug: sig-instrumentation-spotlight-2023
+date: 2023-01-31
+author: "Imran Noor Mohamed(Delivery Hero)"
+---
+
```
Comment (fsmunoz) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:8]: Add **Author**: \<you\> in the main body.

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+**Elana (EH)**: I think SIG Instrumentation is a really interesting SIG because it poses different kinds of opportunities to get involved than in other SIGs. You don’t have to be a software developer to contribute to our SIG! All of our components and subprojects are focused on better understanding Kubernetes and its performance in production, which allowed me to get involved as one of the few SIG Chairs working as an SRE at that time. I like that we provide opportunities for newcomers to contribute through using, testing, and providing feedback on our subprojects, which is a lower barrier to entry. Because many of these projects are out-of-tree, I think one of our challenges is to figure out what’s in scope for core Kubernetes SIGs instrumentation subprojects, what’s missing, and then fill in the gaps.
+
+## Community and contribution
+
+**Imran (INM)**: Kubernetes values community over products.  Any recommendation for anyone looking into getting involved in SIG Instrumentation work? Where should they start (new contributor-friendly areas within SIG?)
```
Comment (fsmunoz) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:55]: Link to https://github.com/kubernetes/community/tree/master/sig-instrumentation#meetings ?

**Comment by fsmunoz** [comment]
Hi @imnmo , great stuff, thank you for your work on this. I made some suggestions (which are on top on what was already reviewed before), minor stuff (except the Author field).

@jberkus: as @sftim mentioned, SIG Spotlights are considered adequate for both, and I've tried to reflect that (in a generalised way) in the last update to the [blogging guidelines](https://github.com/kubernetes/community/blob/master/communication/contributor-comms/storytelling-resources/blog-guidelines.md)

> 1. Article is just for [k8s.dev](http://k8s.dev/blog): this is when it is relevant for the contributor community, and not necessarily for Kubernetes end-users. An example is an article explaining how to use some specific tool or automation that helps with the Kubernetes development process.
> 2. Article is just for [kubernetes.io](https://kubernetes.io/blog/): when the article targets Kubernetes end-users, and not specifically the contributor community. Examples include most technical articles on Kubernetes features, updates on new features and deprecations, etc.
> 3. Article is relevant for both: sometimes, an article will be relevant to both the Kubernetes end-users, and the contributor community. Examples include interviews with SIGs and WGs, articles on technical aspects that are important for the contributor community, etc.

It also describes that a second PR will be made against the main site. The process assumes that both PRs are opened by the editor (with the author also listed) though.

File: content/en/blog/2023/sig-instrumentation-spotlight.md (hunk 9 above)
Comment (imnmo) [reply content/en/blog/2023/sig-instrumentation-spotlight.md:43]: awesome! Good to plug them

**Comment by imnmo** [comment]
@sftim PTAL

**Comment by sftim** [comment]
I recommend asking #sig-docs-blog (on Slack) for advice on getting the PR opened against k/website, and also for feedback on that PR once it's opened.

**Comment by fsmunoz** [comment]
@imnmo the Comms blogging guideline mentioned above has some guidance on the process, and I can also assist in our Slack channel should it be needed. In general though, for articles which are cross-posted, pretty much all of the review should be done in this PR.

**Comment by imnmo** [comment]
@sftim and @fsmunoz I asked in the slack channel for hints to post in k8s.io

File: content/en/blog/2023/sig-instrumentation-spotlight.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
+---
+layout: blog
+title: "Spotlight on SIG Instrumentation"
+slug: sig-instrumentation-spotlight-2023
+date: 2023-02-03
+author: "Imran Noor Mohamed(Delivery Hero)"
```
Comment (sftim) [inline content/en/blog/2023/sig-instrumentation-spotlight.md:6]:
```suggestion
author: "Imran Noor Mohamed (Delivery Hero)"
```

File: content/en/blog/2023/sig-instrumentation-spotlight.md (hunk 10 above)
Comment (sftim) [reply content/en/blog/2023/sig-instrumentation-spotlight.md:8]:
```
author: "Imran Noor Mohamed(Delivery Hero)"
```
in the front matter is fine - IMO

**Review by sftim — COMMENTED** [review]
Thanks. ~I suggest moving the publication date back a few days as we're short on review cycles.~
~How about the 6th of Feb?~


/hold
pending that [second PR](https://github.com/kubernetes/website/pull/39204) getting updated.

**Comment by sftim** [comment]
I can fix up the punctuation in a follow up PR.

/lgtm
/approve

**Pull request merged**